domain/schema/model/sql/0020-network.sql

```diff
@@ -14,9 +14,304 @@ CREATE UNIQUE INDEX idx_cloud_service_application
+    PRIMARY KEY (net_node_uuid, port)
+);
+
+CREATE TABLE link_layer_device_type (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_link_layer_device_type_name
+ON link_layer_device_type (name);
+
+INSERT INTO link_layer_device_type VALUES
+(0, 'unknown'),
+(1, 'loopback'),
+(2, 'ethernet'),
+(3, '802.1q'),
+(4, 'bond'),
+(5, 'bridge'),
+(6, 'vxlan');
+
+CREATE TABLE virtual_port_type (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_virtual_port_type_name
+ON virtual_port_type (name);
+
+INSERT INTO virtual_port_type VALUES
+(0, 'nonvirtualport'),
+(1, 'openvswitch');
+
+CREATE TABLE link_layer_device (
+    uuid TEXT NOT NULL PRIMARY KEY,
+    net_node_uuid TEXT NOT NULL,
+    name TEXT NOT NULL,
+    mtu INT,
+    mac_address TEXT NOT NULL,
+    -- provider_id comes from the provider, no FK.
+    provider_id TEXT,
+    -- one of loopback, ethernet, bridge etc.
+    device_type_id INT NOT NULL,
+    -- one of nonvirtualport, openvswitch
+    virtual_port_type_id INT NOT NULL,
+    -- true if the device should be activated on boot.
+    is_auto_start BOOLEAN NOT NULL DEFAULT false,
+    -- true when the device is up (enabled).
+    is_enabled BOOLEAN NOT NULL DEFAULT true,
+    -- the link layer device which is the parent to this device.
+    parent_device_uuid TEXT,
+    CONSTRAINT fk_link_layer_device_parent_device
+    FOREIGN KEY (parent_device_uuid)
+    REFERENCES link_layer_device (uuid),
+    CONSTRAINT fk_link_layer_device_net_node
+    FOREIGN KEY (net_node_uuid)
+    REFERENCES net_node (uuid),
+    CONSTRAINT fk_link_layer_device_device_type
+    FOREIGN KEY (device_type_id)
+    REFERENCES link_layer_device_type (id),
+    CONSTRAINT fk_link_layer_device_virtual_port_type
+    FOREIGN KEY (virtual_port_type_id)
+    REFERENCES virtual_port_type (id)
+);
+
+CREATE TABLE link_layer_device_ip_address (
+    device_uuid TEXT NOT NULL,
+    address_uuid TEXT NOT NULL,
+    CONSTRAINT fk_link_layer_device_ip_address_device
+    FOREIGN KEY (device_uuid)
+    REFERENCES link_layer_device (uuid),
+    CONSTRAINT fk_link_layer_device_ip_address_address
+    FOREIGN KEY (address_uuid)
+    REFERENCES ip_address (uuid),
+    PRIMARY KEY (device_uuid, address_uuid)
+);
+
+-- ip_address_type represents the possible ways of specifying
+-- an address, either a hostname resolvable by dns lookup,
+-- or IPv4 or IPv6 address.
+CREATE TABLE ip_address_type (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_ip_address_type_name
+ON ip_address_type (name);
+
+INSERT INTO ip_address_type VALUES
+(0, 'ipv4'),
+(1, 'ipv6');
+
+-- ip_address_origin represents the authoritative source of
+-- an ip address.
+CREATE TABLE ip_address_origin (
+    id INT PRIMARY KEY,
+    name TEXT NOT NULL
+);
+
+CREATE UNIQUE INDEX idx_ip_address_origin_name
+ON ip_address_origin (name);
+
+INSERT INTO ip_address_origin VALUES
+(0, 'unknown'),
```

Comment:
I don't think there is ever a situation where we would use _unknown_. If the machine agent reports it, it's _host_, otherwise it'll be _provider_. 

I see you've added this for host names, but I would select `NULL` for the value into the view.

This is better than allowing _unknown_ to end up against an IP address, effectively taking it out of the discovery/subsidence workflow..